### Data Analysis with Panda
- Video Game Sales
- Michael Hendricks
- July 20

In [161]:
import pandas as pd
import numpy as np

In [162]:
df = pd.read_csv('data/vgsales.csv')

In [163]:
df

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

In [164]:
publishers = df.set_index('Publisher')

In [165]:
#publishers

In [166]:
pub_counts = df[['Publisher']].value_counts()
most_common_publisher = pub_counts.max()

In [167]:
#pub_counts.head()


In [168]:
pub_counts.head(1)

Publisher      
Electronic Arts    1351
dtype: int64

### 1. The Most common Video Game Publisher is Electronic arts

In [169]:
plat_counts = df[["Platform"]].value_counts()

In [170]:
#plat_counts.head()

In [171]:
plat_counts.head(1)

Platform
DS          2163
dtype: int64

### 2. The most common platform is DS

In [172]:
genre_counts = df[['Genre']].value_counts()

In [173]:
#genre_counts.head()

In [174]:
genre_counts.head(1)

Genre 
Action    3316
dtype: int64

### 3. The most common genre is Action

In [175]:
name_global_only = df[['Name','Global_Sales']]

### 4. The top 20 highest grossing games are:

In [176]:
name_global_only.head(20)

Name  Global_Sales
0                                     Wii Sports         82.74
1                              Super Mario Bros.         40.24
2                                 Mario Kart Wii         35.82
3                              Wii Sports Resort         33.00
4                       Pokemon Red/Pokemon Blue         31.37
5                                         Tetris         30.26
6                          New Super Mario Bros.         30.01
7                                       Wii Play         29.02
8                      New Super Mario Bros. Wii         28.62
9                                      Duck Hunt         28.31
10                                    Nintendogs         24.76
11                                 Mario Kart DS         23.42
12                   Pokemon Gold/Pokemon Silver         23.10
13                                       Wii Fit         22.72
14                                  Wii Fit Plus         22.00
15                            Kinect Adventures!         21.82
16                            Grand Theft Auto V         21.40
17                 Grand Theft Auto: San Andreas         20.81
18                             Super Mario World         20.61
19  Brain Age: Train Your Brain in Minutes a Day         20.22

In [177]:
name_NA_only = df[['Name','NA_Sales']]

In [178]:
name_NA_only.median()

NA_Sales    0.08
dtype: float64

In [179]:
df[df['NA_Sales'] == 0.08][['Name','NA_Sales']].sort_values('NA_Sales').tail(10)

Name  NA_Sales
8941                              Galerians      0.08
8939   Disney's Donald Duck: Goin' Quackers      0.08
8936                        Toy Story Racer      0.08
8931                   Kung-Fu: High Impact      0.08
8900                           Cardinal Syn      0.08
8887                         Chessmaster II      0.08
8869            T'ai Fu: Wrath of the Tiger      0.08
8860                 Mirror's Edge Catalyst      0.08
8826                       FIFA Soccer 2002      0.08
11492          Ultimate Shooting Collection      0.08

### 5. For North American video game sales, the median is 0.08 million

In [180]:
df[['NA_Sales']].std()

NA_Sales    0.816683
dtype: float64

### 6. The Highest selling video game of all time is 0.8166 standard deviations above the mean

In [181]:
#df[df['Platform'] == 'Wii'][['Name','Platform','Global_Sales']].sort_values('Global_Sales')

In [182]:
wii_games = df[df['Platform'] == 'Wii'][['Name','Platform','Global_Sales']].sort_values('Global_Sales')

In [183]:
#df[df['Platform'] != 'Wii'][['Name','Platform','Global_Sales']].sort_values('Global_Sales')

In [184]:
non_wii_games = df[df['Platform'] != 'Wii'][['Name','Platform','Global_Sales']].sort_values('Global_Sales')

In [185]:
wii_games.mean()

Global_Sales    0.699404
dtype: float64

In [186]:
non_wii_games.mean()

Global_Sales    0.52339
dtype: float64

### 7. Wii games averages 0.699404 million sold per game compared to only 0.52339 million for other platforms

In [187]:
df[['Year']].min()

Year    1980.0
dtype: float64

In [188]:
df[df['Year'] == 1980.0][['Name','Year']].sort_values('Year')

Name    Year
258         Asteroids  1980.0
544   Missile Command  1980.0
1766          Kaboom!  1980.0
1969         Defender  1980.0
2669           Boxing  1980.0
4025       Ice Hockey  1980.0
5366          Freeway  1980.0
6317           Bridge  1980.0
6896         Checkers  1980.0

### 8. The first 9 video games came out in 1980

In [189]:
df[df['Publisher'] == 'Kemco'][['Name','Year']].sort_values('Year').max()

Name    Woody Woodpecker in Crazy Castle 5
Year                                2005.0
dtype: object

### 9. The Last Kemco game was published in 2005

In [190]:
name_global_genre = df[['Name','Global_Sales','Genre']]

In [191]:
name_global_genre.head(20)

Name  Global_Sales         Genre
0                                     Wii Sports         82.74        Sports
1                              Super Mario Bros.         40.24      Platform
2                                 Mario Kart Wii         35.82        Racing
3                              Wii Sports Resort         33.00        Sports
4                       Pokemon Red/Pokemon Blue         31.37  Role-Playing
5                                         Tetris         30.26        Puzzle
6                          New Super Mario Bros.         30.01      Platform
7                                       Wii Play         29.02          Misc
8                      New Super Mario Bros. Wii         28.62      Platform
9                                      Duck Hunt         28.31       Shooter
10                                    Nintendogs         24.76    Simulation
11                                 Mario Kart DS         23.42        Racing
12                   Pokemon Gold/Pokemon Silver         23.10  Role-Playing
13                                       Wii Fit         22.72        Sports
14                                  Wii Fit Plus         22.00        Sports
15                            Kinect Adventures!         21.82          Misc
16                            Grand Theft Auto V         21.40        Action
17                 Grand Theft Auto: San Andreas         20.81        Action
18                             Super Mario World         20.61      Platform
19  Brain Age: Train Your Brain in Minutes a Day         20.22          Misc

### 10. Tetris is the only 'Puzzle' game in the 7op 20 grossing games